src/evolve.js

```diff
@@ -17,10 +18,22 @@ var mapObjIndexed = require('./mapObjIndexed');
  * @return {Object} The transformed object.
  * @example
  *
- *      R.evolve({ elapsed: R.add(1), remaining: R.add(-1) }, { name: 'Tomato', elapsed: 100, remaining: 1400 }); //=> { name: 'Tomato', elapsed: 101, remaining: 1399 }
+ *      var tomato  = {firstName: '  Tomato ', elapsed: 100, remaining: 1400};
+ *      var transformations = {
+ *        firstName: R.trim,
+ *        lastName: R.trim, // Will not get invoked.
+ *        data: {elapsed: R.add(1), remaining: R.add(-1)}
+ *      };
+ *      R.evolve(transformations, tomato); //=> {firstName: 'Tomato', data: {elapsed: 101, remaining: 1399}}
  */
 module.exports = _curry2(function evolve(transformations, object) {
-  return _extend(_extend({}, object), mapObjIndexed(function(fn, key) {
-    return fn(object[key]);
-  }, transformations));
+  var transformation, key, type, result = {};
+  for (key in object) {
+    transformation = transformations[key];
+    type = typeof transformation;
+    result[key] = type === 'function' ? transformation(object[key])
+                : type === 'object'   ? evolve(transformations[key], object[key])
+                                      : object[key];
```

Comment:
Why do we support this case? It seems to me every leaf node of `transformations` should be a function. One should use `{foo: R.always(42)}` rather than `{foo: 42}` if one wishes to transform the value of the `foo` property to `42` in every case.